### PR TITLE
feature(navbar): Unlock listing endpoint to display org counts #342

### PR DESCRIPTION
**Pull request by blaino**
This is so that users can see organization counts in the Organizations pull down.

### CODE REVIEW

**Comment by blaino** [comment]
@arw180 please have a look when you get a chance. I imagine that even if it's ok we won't actually merge until we have had some customer buy-in.

**Comment by rpokorny** [comment]
Be sure to filter out non-approved and non-enabled listings for plain users.

**Comment by rpokorny** [comment]
Actually, the getAllMatchingParams function's primary purpose is to return listing details - the counts are secondary.  As such, using that function just to get counts is a substantial waste of bandwidth at all layers of the system, especially since I assume this will be called every time the discovery page is loaded.  

A simple, better way to do this would be to expose getCountsMatchingParams as a REST endpoint and to call that directly.  

A less simple, but even better way would be to get the information from Elasticsearch, which is already tailored to work only over approved, enabled, listings, and  which is already being queried for the `/storefront` REST endpoint anyway.

**Comment by blaino** [comment]
These are some good ideas - I'm out for the rest of the week and will dive into them when I get back. 

As you know, we are working on a new backend, and, as such, we hoped to change the existing backend as little as possible - using getAllMatchingParams seemed like a way to achieve this goal. But, if it's easy to do the more performant thing (mind you, we are caching the counts on the front end so it isn't that bad), then we should do it.

**Comment by blaino** [comment]
We went with the "simple, better way" Ross listed above (not the "less simple even better way"): expose getCountsMatchingParams as an endpoint. Roles protection to getAllMatchingParams() has been restored.

**Comment by blaino** [comment]
@wski, can I get you to take a first look?

File: src/groovy/marketplace/rest/service/ListingRestService.groovy [hunk 1]
```diff
@@ -35,6 +37,7 @@ class ListingRestService extends RestService<Listing> {
     @Autowired ProfileRestService profileRestService
     @Autowired ListingActivityInternalService listingActivityInternalService
     @Autowired ImageRestService imageRestService
+    @Autowired AgencyRestService agencyRestService
```
Comment (rpokorny) [inline src/groovy/marketplace/rest/service/ListingRestService.groovy]: is this variable used anywhere?

**Comment by wski** [comment]
:+1:

File: src/groovy/marketplace/rest/service/ListingRestService.groovy (hunk 1 above)
Comment (arw180) [inline src/groovy/marketplace/rest/service/ListingRestService.groovy]: ah yes, that can go now. I didn't realize you could pass in `null` for the agencies and have that act as 'all' - I changed it but forgot to remove the associated import

**Comment by blaino** [comment]
@clarkjefcoat Can I get you to review this? (Goes with https://github.com/ozone-development/ozp-center/pull/353)

**Comment by clarkjefcoat** [comment]
:+1:

**Pull request merged**